src/EditorFeatures/Core.Wpf/SignatureHelp/Controller.cs

```diff
@@ -33,6 +33,7 @@ internal partial class Controller :
         private readonly IList<Lazy<ISignatureHelpProvider, OrderableLanguageMetadata>> _allProviders;
         private ImmutableArray<ISignatureHelpProvider> _providers;
         private IContentType _lastSeenContentType;
+        private bool _caretMoveFromArrowKeys;
```

Comment:
mutable state is a bit concerning here.  shoudl htis be stored as state, or passed along as a value?